cpp/benchmarks/string/split.cpp

```diff
@@ -25,63 +23,49 @@
-static void BM_split(benchmark::State& state, split_type rt)
-{
-  cudf::size_type const n_rows{static_cast<cudf::size_type>(state.range(0))};
-  cudf::size_type const max_str_length{static_cast<cudf::size_type>(state.range(1))};
   data_profile const profile = data_profile_builder().distribution(
-    cudf::type_id::STRING, distribution_id::NORMAL, 0, max_str_length);
-  auto const column = create_random_column(cudf::type_id::STRING, row_count{n_rows}, profile);
+    cudf::type_id::STRING, distribution_id::NORMAL, 0, row_width);
+  auto const column = create_random_column(cudf::type_id::STRING, row_count{num_rows}, profile);
   cudf::strings_column_view input(column->view());
   cudf::string_scalar target("+");
 
-  for (auto _ : state) {
-    cuda_event_timer raii(state, true, cudf::get_default_stream());
-    switch (rt) {
-      case split: cudf::strings::split(input, target); break;
-      case split_ws: cudf::strings::split(input); break;
-      case record: cudf::strings::split_record(input, target); break;
-      case record_ws: cudf::strings::split_record(input); break;
-    }
-  }
-
-  state.SetBytesProcessed(state.iterations() * input.chars_size());
-}
+  state.set_cuda_stream(nvbench::make_cuda_stream_view(cudf::get_default_stream().value()));
+  // gather some throughput statistics as well
+  auto chars_size = input.chars_size();
+  state.add_element_count(chars_size, "chars_size");            // number of bytes;
+  state.add_global_memory_reads<nvbench::int8_t>(chars_size);   // all bytes are read;
+  state.add_global_memory_writes<nvbench::int8_t>(chars_size);  // all bytes are written
 
-static void generate_bench_args(benchmark::internal::Benchmark* b)
-{
-  int constexpr min_rows   = 1 << 12;
-  int constexpr max_rows   = 1 << 24;
-  int constexpr row_mult   = 8;
-  int constexpr min_rowlen = 1 << 5;
-  int constexpr max_rowlen = 1 << 13;
-  int constexpr len_mult   = 2;
-  for (int row_count = min_rows; row_count <= max_rows; row_count *= row_mult) {
-    for (int rowlen = min_rowlen; rowlen <= max_rowlen; rowlen *= len_mult) {
-      // avoid generating combinations that exceed the cudf column limit
-      size_t total_chars = static_cast<size_t>(row_count) * rowlen;
-      if (total_chars < static_cast<size_t>(std::numeric_limits<cudf::size_type>::max())) {
-        b->Args({row_count, rowlen});
-      }
-    }
+  if (stype == "split") {
+    state.exec(nvbench::exec_tag::sync,
+               [&](nvbench::launch& launch) { cudf::strings::split(input, target); });
+  } else if (stype == "split_ws") {
+    state.exec(nvbench::exec_tag::sync,
+               [&](nvbench::launch& launch) { cudf::strings::split(input); });
+  } else if (stype == "record") {
+    state.exec(nvbench::exec_tag::sync,
+               [&](nvbench::launch& launch) { cudf::strings::split_record(input, target); });
+  } else {
+    state.exec(nvbench::exec_tag::sync,
+               [&](nvbench::launch& launch) { cudf::strings::split_record(input); });
```

Comment:
Nit: We can use just one `state.exec()` call with the `if/else` statements inside its lambda.

Reply:
I don't want to include the if-statement string compares in the benchmark measurements.